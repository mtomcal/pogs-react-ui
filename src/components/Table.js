import React, {
  Component,
  PropTypes,
} from 'react';

import _ from 'lodash';

function TableRow({headers, fields}) {
  return (
    <tr>
      {headers.map(function toField(header, index) {
        return <td key={`field-${index}`}>{fields[header]}</td>
      })}
    </tr>
  );
}

TableRow.propTypes = {};

function TableHeader({headers}) {
  return (
    <thead>
    <tr>
      {headers.map(function header(headerField, index) {
        return <th key={`header-${index}`}>{_.capitalize(headerField)}</th>
      })}
    </tr>
    </thead>
  );
}

TableHeader.propTypes = {};

class Table extends Component {
  constructor(props) {
    super(props);
  }
  render() {
    const {data, headers} = this.props;

    return (
      <table className="striped">
        <TableHeader headers={headers} />
        <tbody>
        {data.map(function tableRowMap(row, index) {
          return <TableRow key={`row-${index}`} headers={headers} fields={row} />
        })}
        </tbody>
      </table>
    );
  }
}

Table.propTypes = {
  data: PropTypes.arrayOf(PropTypes.object).isRequired
};

export default Table;












