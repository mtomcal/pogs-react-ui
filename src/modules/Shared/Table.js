import React, { Component } from 'react';
import _ from 'lodash';
import PropTypes from 'prop-types';

function TableRow({ headers, fields, unSafe }) {
  return (
    <tr>
      {headers.map(function toField(header, index) {
        const value = fields[header];
        if (unSafe) {
          return (
            <td
              key={`field-${index}`}
              dangerouslySetInnerHTML={{ __html: value }}
            />
          );
        }
        return <td key={`field-${index}`}>{value}</td>;
      })}
    </tr>
  );
}

TableRow.propTypes = {};

function TableHeader({ headers }) {
  return (
    <thead>
      <tr>
        {headers.map(function header(headerField, index) {
          return <th key={`header-${index}`}>{_.capitalize(headerField)}</th>;
        })}
      </tr>
    </thead>
  );
}

TableHeader.propTypes = {};

class Table extends Component {
  render() {
    const { data, headers, unSafe } = this.props;

    return (
      <table className="table table-striped">
        <TableHeader headers={headers} />
        <tbody>
          {data.map(function tableRowMap(row, index) {
            return (
              <TableRow
                key={`row-${index}`}
                headers={headers}
                unSafe={unSafe}
                fields={row}
              />
            );
          })}
        </tbody>
      </table>
    );
  }
}

Table.propTypes = {
  data: PropTypes.arrayOf(PropTypes.object).isRequired,
  unSafe: PropTypes.bool,
};

Table.defaultProps = {
  unSafe: false,
};

export default Table;
