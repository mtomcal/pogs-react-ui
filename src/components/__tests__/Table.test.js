import React from 'react';
import Table from '../Table';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';

function getFakeData() {
  return [
    {id: '1234', description: 'This is cool'},
    {id: '5678', description: 'This is cooler'}
  ];
}

it('renders with two rows and a header', () => {
  const component = renderer.create(
    <Table headers={['id', 'description']} data={getFakeData()} />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
