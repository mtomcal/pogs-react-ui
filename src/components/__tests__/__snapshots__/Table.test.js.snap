exports[`test renders with two rows and a header 1`] = `
<table
  className="table table-striped">
  <thead>
    <tr>
      <th>
        Id
      </th>
      <th>
        Description
      </th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>
        1234
      </td>
      <td>
        This is cool
      </td>
    </tr>
    <tr>
      <td>
        5678
      </td>
      <td>
        This is cooler
      </td>
    </tr>
  </tbody>
</table>
`;
